### PR TITLE
Fix issue 10678

### PR DESCRIPTION
**Pull request by rchavik**
See #10678

### CODE REVIEW

File: src/Routing/RouteBuilder.php [hunk 1]
```diff
@@ -612,7 +612,12 @@ public function prefix($name, $params = [], callable $callback = null)
             $params = [];
         }
         $name = Inflector::underscore($name);
-        $path = '/' . $name;
+        if (isset($params['path'])) {
+            $path = '/' . $params['path'];
```
Comment (ADmad) [inline src/Routing/RouteBuilder.php]: The slash should be added only if doesn't already exist. For e.g. my plugin already sets the path with starting slash https://github.com/ADmad/CakePHP-HybridAuth/blob/master/config/routes.php#L13

**Review by ADmad — CHANGES_REQUESTED** [review]

File: src/Routing/RouteBuilder.php [hunk 2]
```diff
@@ -612,7 +612,12 @@ public function prefix($name, $params = [], callable $callback = null)
             $params = [];
         }
         $name = Inflector::underscore($name);
-        $path = '/' . $name;
+        if (isset($params['path'])) {
+            $path = ($params['path'][0] === '/' ? '' : '/') . $params['path'];
```
Comment (rchavik) [inline src/Routing/RouteBuilder.php]: too convoluted?

Comment (ADmad) [inline src/Routing/RouteBuilder.php]: Maybe `$path = $params['path'][0] !== '/' ?: '/' . $params['path']`?

Comment (rchavik) [inline src/Routing/RouteBuilder.php]: hmm, wouldn't that cause `true` (casted to `1`) to be prepended?

Comment (rchavik) [inline src/Routing/RouteBuilder.php]: $ git diff

```diff
diff --git a/src/Routing/RouteBuilder.php b/src/Routing/RouteBuilder.php
index a94d27f..11f5417 100644
--- a/src/Routing/RouteBuilder.php
+++ b/src/Routing/RouteBuilder.php
@@ -613,7 +613,9 @@ class RouteBuilder
         }
         $name = Inflector::underscore($name);
         if (isset($params['path'])) {
-            $path = ($params['path'][0] === '/' ? '' : '/') . $params['path'];
+            debug($params['path']);
+            $path = $params['path'][0] !== '/' ?: '/' . $params['path'];
+            debug($path);
             unset($params['path']);
         } else {
             $path = '/' . $name;
```

```
$ vendor/bin/phpunit --stderr --filter DotIn 
PHPUnit 5.7.20 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)/src/Routing/RouteBuilder.php (line 616)
########## DEBUG ##########
'v1.0'
###########################
/src/Routing/RouteBuilder.php (line 618)
########## DEBUG ##########
true
###########################


Time: 896 ms, Memory: 96.75MB

There was 1 failure:

1) Cake\Test\TestCase\Routing\RouteBuilderTest::testPathWithDotInPrefix
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'/admin/api/v1.0'
+'/admin/api1'

/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/tests/TestCase/Routing/RouteBuilderTest.php:363
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/src/Routing/RouteBuilder.php:701
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/src/Routing/RouteBuilder.php:627
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/tests/TestCase/Routing/RouteBuilderTest.php:369
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/src/Routing/RouteBuilder.php:701
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/src/Routing/RouteBuilder.php:627
/home/rachman/sites/croogo3.dev/vendor/cakephp/cakephp/tests/TestCase/Routing/RouteBuilderTest.php:370

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

File: src/Routing/RouteBuilder.php [hunk 3]
```diff
@@ -612,7 +612,12 @@ public function prefix($name, $params = [], callable $callback = null)
             $params = [];
         }
         $name = Inflector::underscore($name);
-        $path = '/' . $name;
+        if (isset($params['path'])) {
```
Comment (markstory) [inline src/Routing/RouteBuilder.php]: What if its `false`. The next line will fail.

File: src/Routing/RouteBuilder.php (hunk 2 above)
Comment (markstory) [inline src/Routing/RouteBuilder.php]: If the developer is passing a path, they can just do the right thing. I don't think we should be forcing a `/` on people.

File: src/Routing/RouteBuilder.php [hunk 4]
```diff
@@ -613,6 +613,10 @@ public function prefix($name, $params = [], callable $callback = null)
         }
         $name = Inflector::underscore($name);
         $path = '/' . $name;
+        if (isset($params['path'])) {
+            $path = $params['path'];
+            unset($params['path']);
```
Comment (markstory) [inline src/Routing/RouteBuilder.php]: :+1: much simpler now.

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/Routing/RouteBuilder.php (hunk 4 above)
Comment (markstory) [inline src/Routing/RouteBuilder.php]: This new option should be added to the doc block.

**Review by markstory — CHANGES_REQUESTED** [review]
API docs need to be updated. Sorry for missing that earlier.

**Review by markstory — APPROVED** [review]

**Pull request merged**